CRM/Dedupe/Merger.php

```diff
@@ -1696,12 +1706,12 @@ public static function moveAllBelongings($mainId, $otherId, $migrationInfo, $che
     // move view only custom fields CRM-5362
     $viewOnlyCustomFields = array();
     foreach ($submitted as $key => $value) {
-      $fid = (int) substr($key, 7);
-      if (array_key_exists($fid, $cFields) && !empty($cFields[$fid]['attributes']['is_view'])) {
+      $fid = CRM_Core_BAO_CustomField::getKeyID($key);
```

Comment:
this just prevents any other field (I can think of no examples) that is not a custom field being mistaken for one. uses regex not 'is last character a number'